Microsoft.Alm.Git/GitInstallation.cs

```diff
@@ -64,6 +66,27 @@ internal GitInstallation(string path, KnownGitDistribution version)
             Debug.Assert(CommonLibexecPaths.ContainsKey(version), "The `version` parameter not found in `CommonLibExecPaths`.");
             Debug.Assert(CommonShPaths.ContainsKey(version), "The `version` parameter not found in `CommonShPaths`.");
 
+            path = path.TrimEnd('\\');
+
+            // Make sure the GitExeName isn't included as a part of the path.
+            if (path.ToLower().EndsWith(AllVersionGitPath))
```

Comment:
Please use case-insensitive string comparer instead of allocating a new lower-case string to perform comparisons.